app/code/Magento/Developer/Model/XmlCatalog/Format/PhpStorm.php

```diff
@@ -23,20 +23,28 @@ class PhpStorm implements FormatInterface
     private $currentDirRead;
 
     /**
-     * @var \Magento\Framework\Filesystem\File\WriteFactory
+     * @var FileWriteFactory
      */
     private $fileWriteFactory;
 
     /**
+     * @var DomDocumentFactory
+     */
+    private $domDocumentFactory;
+
+    /**
      * @param ReadFactory $readFactory
-     * @param \Magento\Framework\Filesystem\File\WriteFactory $fileWriteFactory
+     * @param FileWriteFactory $fileWriteFactory
+     * @param DomDocumentFactory $domDocumentFactory
      */
     public function __construct(
         ReadFactory $readFactory,
-        \Magento\Framework\Filesystem\File\WriteFactory $fileWriteFactory
+        FileWriteFactory $fileWriteFactory,
+        DomDocumentFactory $domDocumentFactory
```

Comment:
I was already wondering how BC was being handled! I'll make the constructor parameter optional for BC.

Reply:
It might also be a good idea to revert the change made to `\Magento\Framework\Filesystem\File\WriteFactory $fileWriteFactory` as this is not specific to your change and I am not 100% sure how BC would work in this case.